### ISSUE TITLE
Endless loop when setting up bi-directional replication

### ISSUE THREAD

**Issue by tanago** [2ndQuadrant/pglogical#210]
We are trying to setup a 2 node bi-directional replication. dbA <----> dbB

After we set it upon both nodes (using last_update_wins conflict resolution)  we check both subscriptions and both are in active state. Then we insert 1 row of data and endless loop of transactions occurs that can be seen in the logs. 

```
pg11_1  | 2019-06-03 07:11:04.752 UTC [115] DEBUG:  CONFLICT: remote INSERT on relation public."DeviceCodes" (local index PK_DeviceCodes). Resolution: apply_remote.
pg11_1  | 2019-06-03 07:11:04.752 UTC [115] DETAIL:  
existing local tuple 
{
DeviceCode[varchar]:a
UserCode[varchar]:a
SubjectId[varchar]:a
ClientId[varchar]:a
Data[varchar]:a
}
xid=613,origin=0,timestamp=2019-06-03 07:11:04.717526+00;
remote tuple
{
DeviceCode[varchar]:a
UserCode[varchar]:a
SubjectId[varchar]:a
ClientId[varchar]:a
Data[varchar]:a
}
in xact origin=1,timestamp=2019-06-03 07:11:04.717526+00,commit_lsn=0/196BC08
```

```
pg11_1  | 2019-06-03 07:11:05.163 UTC [115] CONTEXT:  apply (unknown action) in commit before 0/196BD20, xid 609 committed at 2019-06-03 07:11:04.717526+00 (action #1) from node replorigin 1
pg11_1  | 2019-06-03 07:11:05.163 UTC [115] CONTEXT:  apply COMMIT in commit before 0/196BD68, xid 609 committed at 2019-06-03 07:11:04.717526+00 (action #3) from node replorigin 1
pg11_1  | 2019-06-03 07:11:05.575 UTC [115] CONTEXT:  apply (unknown action) in commit before 0/196BE00, xid 610 committed at 2019-06-03 07:11:04.717526+00 (action #1) from node replorigin 1
pg11_1  | 2019-06-03 07:11:05.576 UTC [115] CONTEXT:  apply COMMIT in commit before 0/196BE48, xid 610 committed at 2019-06-03 07:11:04.717526+00 (action #3) from node replorigin 1
pg11_1  | 2019-06-03 07:11:05.978 UTC [115] CONTEXT:  apply (unknown action) in commit before 0/196BEE0, xid 611 committed at 2019-06-03 07:11:04.717526+00 (action #1) from node replorigin 1
pg11_1  | 2019-06-03 07:11:05.978 UTC [115] CONTEXT:  apply COMMIT in commit before 0/196BF28, xid 611 committed at 2019-06-03 07:11:04.717526+00 (action #3) from node replorigin 1
pg11_1  | 2019-06-03 07:11:06.408 UTC [115] CONTEXT:  apply (unknown action) in commit before 0/196BFC0, xid 612 committed at 2019-06-03 07:11:04.717526+00 (action #1) from node replorigin 1
pg11_1  | 2019-06-03 07:11:06.408 UTC [115] CONTEXT:  apply COMMIT in commit before 0/196C020, xid 612 committed at 2019-06-03 07:11:04.717526+00 (action #3) from node replorigin 1
```

as you can see transaction ID is increasing +1, +1, +1 for the same record that has timestamp 2019-06-03 07:11:04.717526+00

So record with ID 1 is inserted into dbA and gets replicated into dbB. dbB returns the record into dbA that returns it to dbB to dbA and that cycle never ends.
The bug (or feature) comes from https://github.com/2ndQuadrant/pglogical/blob/REL2_x_STABLE/pglogical_conflict.c#L427-L438

If the else block returns KeepLocal resolution the endless loop will stop because "cmp" equals 0 in such cases where it's comparing records with identical timestamps

**Comment by PJMODOS** [comment]
You probably want to use `forward_origins := '{}'` when doing `create_subscription`, that will prevent loop. See readme for details of that argument.

**Comment by tanago** [comment]
> You probably want to use `forward_origins := '{}'` when doing `create_subscription`, that will prevent loop. See readme for details of that argument.

Yea I was playing around with a few properties and tried this but with syntax `forward_origins := ARRAY['']` that got me an error 

> empty array not allowed, only 'all'

which kinda contradicts with the documentation: 

> [...] currently only supported values are empty array meaning don’t forward any changes that didn’t originate [...]

and I was about to write another bug, haha

but with `forward_origins := '{}'` it worked like charm and replication issues are fixed.

tl;dr thanks, works!

**Comment by vyruss** [comment]
@PJMODOS many thanks for the quick response, Petr!

**Comment by saideepakb** [comment]
Hi @tanago @vyruss do you remember if `forward_origins := '{}'` is to be set on both subscriptions?
